components/canvas/ProcessHierarchy/useProcessHierarchyTabs.tsx

```diff
@@ -0,0 +1,65 @@
+import { useRouter } from "next/router";
+import { useEffect, useState } from "react";
+
+export const useProcessHierarchyTabs = () => {
+  const { query, pathname, push } = useRouter();
+
+  const [activeTab, setActiveTab] = useState(Number(query.tab) || 0);
+  const [showNewConnectionForm, setShowNewConnectionForm] = useState(
+    query.showNewConnection === "true"
+  );
```

Comment:
This can be refactored into a `useMemo`
```ts
const showNewConnectionForm = useMemo(() => query.showNewConnectionForm === "true", [query.showNewConnectionForm])
```

Or even better, since the calculation is so small, I believe it would be more performant to not use any hooks at all.

```ts
const showNewConnectionForm = query.showNewConnectionForm === "true"
```

Regardless of which solution you choose, you would get rid of a `useState` and a `useEffect`